### PR TITLE
Added order related logic: entity, dao, service and rest

### PR DESCRIPTION
**Pull request by marinakrivonosova**
order service and rest logic added
product dao logic needed for order service layer added
line items dao logic added

### CODE REVIEW

File: src/main/java/orders/OrderService.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    @Autowired
+    public OrderService(final OrderDao orderDao, final IdGenerator idGenerator, final LineItemDao lineItemDao, final ProductDao productDao, final Configuration configuration) {
+        this.orderDao = orderDao;
+        this.idGenerator = idGenerator;
+        this.lineItemDao = lineItemDao;
+        this.productDao = productDao;
+        this.configuration = configuration;
+    }
+
+    public String addOrder(final String userId, final List<CreateLineItem> createLineItemList, final String address, final String deliveryMethodId, final String paymentMethodId) {
```
Comment (marinakrivonosova) [inline src/main/java/orders/OrderService.java]: will add transaction support in next commits

File: src/main/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+        <!-- encoders are assigned the type
+             ch.qos.logback.classic.encoder.PatternLayoutEncoder by default -->
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="debug">
```
Comment (marinakrivonosova) [inline src/main/resources/logback.xml]: example from logback doc, will update level after logging set up

File: pom.xml [hunk 3]
```diff
@@ -24,6 +36,17 @@
             <artifactId>spring-jdbc</artifactId>
             <version>${spring.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-beans</artifactId>
+            <version>${spring.version}</version>
+        </dependency>
+        <dependency>
```
Comment (LebedkoDmitry) [inline pom.xml]: Won't the `spring-beans` and the `spring-context` dependencies be added transitively? Do we really have to declare them explicitly?

File: pom.xml [hunk 4]
```diff
@@ -34,6 +57,23 @@
             <artifactId>liquibase-core</artifactId>
             <version>3.8.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.28</version>
```
Comment (LebedkoDmitry) [inline pom.xml]: The version has to be declared as a property.

File: pom.xml [hunk 5]
```diff
@@ -34,6 +57,23 @@
...
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.3</version>
```
Comment (LebedkoDmitry) [inline pom.xml]: The version has to be declared as a property.

File: pom.xml [hunk 6]
```diff
@@ -34,6 +57,23 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+            <version>4.0.1</version>
```
Comment (LebedkoDmitry) [inline pom.xml]: The version has to be declared as a property.

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+initial.orderStatusId=osId1
```
Comment (LebedkoDmitry) [inline src/main/resources/application.properties]: I'd better set names of the initial statuses like `initial.orderStatus=completed` and so on.
The intention was to decouple the initial values and all available statuses.

**Pull request merged**